### PR TITLE
Glob import support

### PR DESCRIPTION
**Pull request by bioball**
This adds support for glob imports!

This PR is based on top of https://github.com/apple/pkl-lsp/pull/11, and should be reviewed after that other one gets merged.

### CODE REVIEW

File: src/main/kotlin/org/pkl/lsp/services/PklProjectManager.kt [hunk 1]
```diff
@@ -75,15 +75,15 @@ class PklProjectManager(project: Project) : Component(project) {
     if (file !is FsFile) null else pklProjects[file.projectKey]
 
   /** Tracks when PklProject files get added or deleted from the workspace. */
-  val addedOrRemovedFilesModificationTracker = SimpleModificationTracker()
+  val addedOrRemovedFilesModificationTracker =
+    project.pklFileTracker.filter { event -> event.files.any { it.path.endsWith("PklProject") } }
```
Comment (stackoverflow) [inline src/main/kotlin/org/pkl/lsp/services/PklProjectManager.kt:79]: This is not related to glob support, right?

**Review by stackoverflow — APPROVED** [review]
Just some small comments. Approving to unblock.

File: src/main/kotlin/org/pkl/lsp/features/GoToDefinitionFeature.kt [hunk 2]
```diff
@@ -31,40 +32,78 @@ import org.pkl.lsp.type.computeThisType
...
     }
     return server.builder().runningBuild(params.textDocument.uri).thenApply(::run)
   }
 
-  private fun resolveDeclaration(
+  private fun resolveModuleUri(
+    originalNode: PklStringConstant,
+    context: PklProject?,
+  ): List<LocationLink> {
+    val parent = originalNode.parent as PklModuleUriOwner
```
[the remark this answers is not in the document]
Comment (bioball) [reply src/main/kotlin/org/pkl/lsp/features/GoToDefinitionFeature.kt:68]: Good point!

File: src/main/kotlin/org/pkl/lsp/services/PklProjectManager.kt (hunk 1 above)
Comment (bioball) [reply src/main/kotlin/org/pkl/lsp/services/PklProjectManager.kt:79]: This is only related in that I needed to introduce another file tracker that just tracked when Pkl files get added or removed from the workspace. We already have other trackers for added/removed files, so I introduced a tracker just for adding/removing Pkl files, and made this one simply derived from it.
